src/main/java/com/google/sps/servlets/EditOrganizationServlet.java

```diff
@@ -18,10 +18,68 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Arrays;
+import java.text.SimpleDateFormat;
 import com.google.sps.data.User;
+import com.google.sps.data.HistoryManager;
+import com.google.sps.data.OrganizationUpdater;
+import com.google.sps.data.GivrUser;
+import java.time.Instant;
 import java.io.IOException;
+import com.google.gson.Gson;
 
 @WebServlet("/edit-organization")
 public class EditOrganizationServlet extends HttpServlet {
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+    //Get the proper organization entity for updating
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    long organizationId = Long.parseLong(request.getParameter("id"));
+    Key organizationKey = KeyFactory.createKey("Distributor", organizationId);
+    Entity organizationEntity = new Entity("Distributor");
+
+    // try catch for compliation purposes, servlet will not be called without a valid id param
+    try {
+      organizationEntity = datastore.get(organizationKey);
+    } catch(com.google.appengine.api.datastore.EntityNotFoundException err) {
+        response.sendError(HttpServletResponse.SC_NOT_FOUND);
+        return;
+    }
+    
+    GivrUser user = GivrUser.getLoggedInUser();
+
+    OrganizationUpdater organizationUpdater = new OrganizationUpdater(organizationEntity);
+    
+    try {
+      organizationUpdater.updateOrganization(request, organizationId, user, /*forRegistration*/ false);
+    } catch(IllegalArgumentException err) {
+        response.sendError(HttpServletResponse.SC_NOT_FOUND);
+        return;
+    }
+
+    // updates entity with changed properties from the form
+    organizationEntity = organizationUpdater.getEntity();
+
+    //TODO: get timestamp with transactions instead
+    long millisecondSinceEpoch = Instant.now().toEpochMilli();
+    organizationEntity.setProperty("lastEditTimeStampMillis", millisecondSinceEpoch);
+
+    ArrayList<EmbeddedEntity> changeHistory = (ArrayList) organizationEntity.getProperty("changeHistory");
```

Comment:
Suggestion: if updateOrganization accepted the result from history.recordHistory() you can keep all the update logic in OrganizationUpdater. Thoughts?

Reply:
The way I envisioned the Updater was that it was primarily parse form input from the user. We could definitely move all the update logic to the end of it but I think we should essentially move lines 73-79 to the end of it as well as modify it to handle what the history would be when addorganizationservlet calls it as well as set the creation timestamp. Your thoughts?

Reply:
It both parses and stores all the changes you want on the entity, right? Seems like 73-79 would be done in both edit and add org (and also in update good, etc). Doing it in another PR sounds fine, but file an issue to track it.